portfolio/src/main/java/com/google/sps/servlets/CreateNewComment.java

```diff
@@ -31,17 +33,35 @@
  */
 @WebServlet("/create-new-comment")
 public class CreateNewComment extends HttpServlet {  
-  
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String username = request.getParameter("username"); 
-    String commentContent = request.getParameter("comment"); 
 
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html"); // set response type
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String userEmail = userService.getCurrentUser().getEmail();
+      response.getWriter().println("Hello " + userEmail + "! Now that you're logged in, feel free to leave a comment.");   
+    } else {
+      response.getWriter().println("Please log in to leave a comment.");   
+    }
+  }
 
-    Entity commentEntity = Comment.createNewCommentEntity(username, commentContent);
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); // create instance of DatastoreService class
-    datastore.put(commentEntity);
+  
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Only logged-in users can add a comment
+    UserService userService = UserServiceFactory.getUserService();
+    if (userService.isUserLoggedIn()) {
+      String username = request.getParameter("username"); 
+      String commentContent = request.getParameter("comment"); 
+      response.setContentType("text/html"); 
+      Entity commentEntity = Comment.createNewCommentEntity(username, commentContent);
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService(); // create instance of DatastoreService class
+      datastore.put(commentEntity);
+    } else {
+      response.setStatus(401); // 401 specifies unauthorized/unauthenticated  
```

Comment:
Are both of these required?  The fact that 401 is a parameter for both leads me to suspect that the first may be redundant.  Please verify.

Reply:
I think that's right. I've removed the first one